### DISCUSSION TITLE
Making xarray_plots better

### DISCUSSION THREAD

**Discussion by blychs** [NOAA-CSL/MELODIES-MONET#303] [Ideas]
This is for developing xarray_plots.
After discussing with @mbruckner-work , this is what we came up to (other than some bugfixes):
a) We need to add map projections to the mapping functions
b) Weight averages by area (for structured grids, using the cosine should do the trick)
c) Adding scatter density plots
d) Adding zonal means
e) pcolormesh option for level 3 data (currently, it is using scatter plots)
f) Updating the docs!

Please add here any other ideas.

**Comment by mbruckner-work** [comment]
While working on updating maps found code that may already do point a. If so, we need to document how to use this and probably add a transform to the scatter/pcolormesh generation.

In make_spatial_bias_gridded we have: 

```
    #Add options that this could be included in the fig_kwargs in yaml file too.
    if 'extent' not in map_kwargs:
        map_kwargs['extent'] = [lonmin,lonmax,latmin,latmax] 
    if 'crs' not in map_kwargs:
        map_kwargs['crs'] = proj
    
```

**Reply by blychs** [reply]
Yes, the main issue with this approach is that it still assumes that it is not crossing the antimeridian.
A simple workaround might be creating a function to select that, but I guess that that would require rotating the map?

**Reply by mbruckner-work** [reply]
Mh, yeah we might need to make some changes to properly set the extent for different projections. Although this may already be okay because the extent is set with:  ``` ax.axes.set_extent(map_kwargs['extent'],crs=ccrs.PlateCarree()) ```
It doesn't look like right now the projection stuff uses anything other than ccrs.PlateCarree() in the satellite maps and I haven't tried to change it yet. 

To get the circular bound to render right for some projections, we'd need to do something like in the cartopy examples (https://scitools.org.uk/cartopy/docs/v0.15/examples/always_circular_stereo.html). 

For the anti-meridian stuff, we might need to use cartopy's util.add_cyclic_point?

**Comment by mbruckner-work** [comment]
Adding usage of xrplots to the driver. 
@blychs, are the boxplot functions any different than the surfplots implementation yet? 

The "make_spatial_dist" function needs a section in the driver. 
Adding an plot group data_proc argument for xarray-based Taylor plots- 'normalization'. Will default to False if not provided.

**Comment by L12D2** [comment]
area_mean() included in this link could address this if it hasn't been dealt with yet. Recently we discussed the weakness of no area weighting within MELODIES MONET. So, seeing this code made me think of this thread.

https://ncar.github.io/gdex-examples/hadisst-elnino/
